tapgarden/planter.go

```diff
@@ -801,6 +805,30 @@ func (c *ChainPlanter) prepAssetSeedling(ctx context.Context,
 	return nil
 }
 
+// updateMintingProofs is called by the re-org watcher when it detects a re-org
+// and has updated the minting proofs. This cannot be done by the caretaker
+// itself, because its job is already done at the point that a re-org can happen
+// (the batch is finalized after a single confirmation).
+func (c *ChainPlanter) updateMintingProofs(proofs []*proof.Proof) error {
+	ctx, cancel := c.WithCtxQuitNoTimeout()
+	defer cancel()
+
+	headerVerifier := GenHeaderVerifier(ctx, c.cfg.ChainBridge)
+	for idx := range proofs {
+		p := proofs[idx]
+
+		err := proof.ReplaceLastProofInBlob(
```

Comment:
I think there's an edge case here if after we've done the minting, the asset is them immediately transferred after. We'd then have the updated proof, but override the final transfer.

Reply:
Hmm, you're right. We do check that we're trying to update the correct proof (script key check in `ReplaceLastProofInBlob`). But I guess we should instead try to find which proof we need to replace (it might not be the last one in the file) and replace that instead of always assuming it's the last in the file.

Reply:
> an edge case here if after we've done the minting, the asset is them immediately transferred after

I wander if we would encounter a similar problem when sending an asset twice in close succession.

Reply:
Yeah I guess it also depends on behavior re nested zero conf transactions as well. Even if confirmed, eg: minting confirmed in block `A`, then transfer in block `B`, then re-org back to block `Z`. In this case we need to update both of them, but also keep the ordering: `A'`, then `B'` and not the other way around.

Reply:
Assuming the re-org watcher has _both_ requests (so watch for `A` and then `B`), if it groups things by sort of root prev ID (the descendent chain), then it should be able to enforce the ordering of updates.